### DISCUSSION TITLE
Introduce pricing and a robust messaging capability

### DISCUSSION THREAD

**Discussion by DovOps** [finos/traderX#196] [Ideas]
The initial project has a rudimentary publish-subscribe mechanism, based on NodeJS+SocketIO so that it's easy to run the whole TraderX system as a local project with minimal overhead/complexity. This obviously has its limitations, though it does achieve simplicity.  It would be good to implement a more robust messaging capability which could become the successor to SocketIO - offering both service-to-service messaging as well as service-to-browser (e.g. websocket) messaging.   A simple embedded capability which could also be deployed in a more robust manner would provide a better foundation to implement real-time simulated pricing across the TraderX stack - something which is missing from the barebones simple implementation.

**Comment by maoo** [comment]
@DovOps - could this use case be achieved also via FDC3? See https://github.com/finos/traderX/discussions/189

**Reply by DovOps** [reply]
I don't think this is FDC3 relevant as the issue here around messaging is more of the back-end server side plumbing - replacing the trade feed's server-side interactions with Trade Service and Trade Processor, and eventually to support a pricing service/streaming capability.

Yes the trade-feed we have is also the web-socket real-time feed to the GUI, but that's the secondary use case.  Right now, to maintain simplicity, there's a single websocket implementation used for backend and frontend. In a more robust solution there's probably a GRPC/GPB or FIX interaction on th eback, with some sort of front-end adapter for streaming to the gui.  

FDC3 is more relevant on the  GUI side of the equation, but more for interoperating between the GUI and other GUIs, and less for inter-connecting the GUI to the back-end.
